src/jq_test.c

```diff
@@ -490,4 +490,98 @@ static void jv_test() {
     //jv_dump(jv_copy(o2), 0); printf("\n");
     jv_free(o2);
   }
+
+  {
+	assert(jv_equal(
+			jv_paths(
+				jv_parse("{"
+					"\"key\":["
+						"{\"this\":{\"some\":true}},"
+						"false,"
+						"null,"
+						"10,"
+						"[true]"
+					"]"
+				"}")
+			),
+			jv_parse("["
+				"[\"key\"],"
+				"[\"key\", 0],"
+				"[\"key\", 0, \"this\"],"
+				"[\"key\", 0, \"this\", \"some\"],"
+				"[\"key\", 1],"
+				"[\"key\", 2],"
+				"[\"key\", 3],"
+				"[\"key\", 4],"
+				"[\"key\", 4, 0]"
+			"]")
+		));
+
+  }
+
+  {
+	assert(jv_equal(
+			jv_addpath(
+				jv_parse("{"
+					"\"key\":{"
+						"\"some\":{"
+							"\"test\":\"value\""
+						"},"
+						"\"other\":\"thing\""
+					"}"
+				"}"),
+				JV_ARRAY(jv_string("key")),
+				jv_parse("{"
+					"\"some\":{"
+						"\"test\":\"other\""
+					"},"
+					"\"added\":\"thing\""
+				"}")
+			),
+			jv_parse("{"
+				"\"key\":{"
+					"\"some\":{"
+						"\"test\":\"other\""
+					"},"
+					"\"other\":\"thing\","
+					"\"added\":\"thing\""
+				"}"
+			"}")
+		)
+	);
+
+  }
+
+  {
+	jv initial = jv_parse("{\"test\":[{\"some\":\"value\"}, 1, true, false, null]}");
+	jv new = jv_unshare(jv_copy(initial));
+
+	assert(jv_equal(jv_copy(initial), jv_copy(new)));
+	assert(!jv_identical(jv_copy(initial), jv_copy(new)));
+
```

Comment:
It might be good to have a `jv_unshared()` function that returns true if the given value is recursively unshared in that every value down from it has `jv_get_refcount()` of `1`.  Such a `jv_unshared()` function would have to _not_ consume a reference to its argument, naturally, **or** maybe it should check that the top-level value has `2` refcounts and all values below have `1` refcounts.

Then we could use `jv_getpath()` on `initial` to acquire an extra refcount on a value far from the `initial` root, and then we could check that neither `initial` nor that value obtained with `jv_getpath()` are `jv_unshared()`.

Reply:
I like that idea. It would make testing easier as well. (the name might just be a bit too similar so simething like jv_is_unshared() might be better)